grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/GrpcMessageMarshaller.java

```diff
@@ -244,7 +256,12 @@ private <T> Message deserializeProto(PrototypeMarshaller<T> marshaller, ByteBuf
                 stream = CodedInputStream.newInstance(buf.nioBuffer());
             }
             try {
-                final Message msg = prototype.getParserForType().parseFrom(stream);
+                final Message msg;
+                if (useMethodMarshaller) {
+                    msg = (Message) marshaller.parse(new ByteBufInputStream(buf));
+                } else {
+                    msg = prototype.getParserForType().parseFrom(stream);
+                }
                 try {
                     stream.checkLastTagWas(0);
```

Comment:
As @ikhoon pointed out, `stream.checkLastTagWas(0)` isn't necessary and we don't even have to create the `CodedInputStream` when `useMethodMarshaller` is true. We can fix this:
```java
try {
    final Message msg;
    if (useMethodMarshaller) {
        msg = (Message) marshaller.parse(new ByteBufInputStream(buf));
    } else {
        final CodedInputStream stream;
        if (unsafeWrapDeserializedBuffer) {
            stream = UnsafeByteOperations.unsafeWrap(buf.nioBuffer()).newCodedInput();
            stream.enableAliasing(true);
        } else {
            stream = CodedInputStream.newInstance(buf.nioBuffer());
        }
        msg = prototype.getParserForType().parseFrom(stream);
        try {
            stream.checkLastTagWas(0);
        } catch (InvalidProtocolBufferException e) {
            e.setUnfinishedMessage(msg);
            throw e;
        }
    }
    return msg;
} catch (InvalidProtocolBufferException e) {
    throw Status.INTERNAL.withDescription("Invalid protobuf byte sequence")
                         .withCause(e).asRuntimeException();
}
```

Reply:
Since calling `stream.checkLastTagWas(0)` is unnecessary, Can I remove this part of code below?

```java
        try {
            stream.checkLastTagWas(0);
        } catch (InvalidProtocolBufferException e) {
            e.setUnfinishedMessage(msg);
            throw e;
        }
```

Reply:
It's unnecessary only when `useMethodMarshaller == true`. 😉

Reply:
Applied feedback, Thanks.